src/software/sensor_fusion/filter/ball_filter.cpp

```diff
@@ -245,8 +245,17 @@ LinearRegressionResults BallFilter::getLinearRegressionLine(
         A.bdcSvd(Eigen::ComputeThinU | Eigen::ComputeThinV).solve(b);
     // How to calculate the error is from
     // https://eigen.tuxfamily.org/dox/group__TutorialLinearAlgebra.html
-    double regression_error =
-        (A * regression_vector - b).norm() / (b.norm() + 1.0e-9);  // norm() is L2 norm
+    double regression_error;
+
+    if ((A * regression_vector - b).norm() == 0 && b.norm() == 0)
+    {
+        regression_error = 0;
+    }
+    else
+    {
+        regression_error =
+            (A * regression_vector - b).norm() / (b.norm());  // norm() is L2 norm
```

Comment:
looks like it's still possible to divide by zero

Reply:
In the test cases, if `b.norm()` was 0, nominator was 0.
To avoid cases where `b.norm()` is 0 but nominator is not, changed so `if (nominator == 0 || b.norm() == 0) {...`